File: matlab/build_graphs_group.m
function build_graphs_group(data)
%build_graphs_group Визуализация результатов сравнения для всех сценариев
%   Позволяет визуализировать динамическое число сценариев

[N,P] = size(data,1:2);
x = 1:N;

% Среднеквадратическое отклонение
tiledlayout(2,1)
ax1 = nexttile;
for p = 1:P
    plot(x,data(:,p,1),"LineWidth",2.0);
    hold on
end
title('Среднеквадратическое отклонение')
grid on
xlabel('Номер пары изображений','FontWeight','bold')
ylabel('СКО, пикс.','FontWeight','bold')

% Максимум отклонения
ax2 = nexttile;
for p = 1:P
    plot(x,data(:,p,2),"LineWidth",2.0);
    hold on
end
title('Максимум отклонения')
grid on
xlabel('Номер пары изображений','FontWeight','bold')
ylabel('Max, пикс.','FontWeight','bold')

% Формирование легенды
legent_text = strings(P,1);
for p = 1:P
    legent_text(p) = strcat("processing ",string(p));
end
legend(ax1,legent_text,'Location','bestoutside')
legend(ax2,legent_text,'Location','bestoutside')

end

File: src/build_graphs_group.py
import numpy as np
import matplotlib.pyplot as plt

def build_graphs_group(data):
    """
    build_graphs_group Visualization of comparison results for all scenarios
    Allows visualizing a dynamic number of scenarios
    """
    
    N, P = data.shape[:2]
    x = np.arange(1, N + 1)

    # Root mean square deviation
    fig, (ax1, ax2) = plt.subplots(2, 1, figsize=(10, 8))
    for p in range(P):
        ax1.plot(x, data[:, p, 0], linewidth=2.0)
    ax1.set_title('Root Mean Square Deviation')
    ax1.grid(True)
    ax1.set_xlabel('Image Pair Number', fontweight='bold')
    ax1.set_ylabel('RMS, pixels', fontweight='bold')

    # Maximum deviation
    for p in range(P):
        ax2.plot(x, data[:, p, 1], linewidth=2.0)
    ax2.set_title('Maximum Deviation')
    ax2.grid(True)
    ax2.set_xlabel('Image Pair Number', fontweight='bold')
    ax2.set_ylabel('Max, pixels', fontweight='bold')

    # Create legend
    legend_text = [f'processing {p + 1}' for p in range(P)]
    ax1.legend(legend_text, loc='best')
    ax2.legend(legend_text, loc='best')

    plt.tight_layout()
    plt.show()

# Example usage
# data = np.random.rand(10, 5, 2)  # Example data
# build_graphs_group(data)